salt/minion.py

```diff
@@ -3281,20 +3285,14 @@ def destroy(self):
         """
         Tear down the minion
         """
-        if self._running is False:
-            return
-
         self._running = False
         if hasattr(self, "schedule"):
             del self.schedule
         if hasattr(self, "pub_channel") and self.pub_channel is not None:
             self.pub_channel.on_recv(None)
-            if hasattr(self.pub_channel, "close"):
-                self.pub_channel.close()
-            del self.pub_channel
-        if hasattr(self, "req_channel") and self.req_channel:
+            self.pub_channel.close()
```

Comment:
```suggestion
            self.pub_channel.close()
            self.pub_channel = None
```
?
Since right above this we also check to see if `self.pub_channel is not None`?

Reply:
Setting these attributes back to None serves no good purpose because we are no longer relying on garbage collection for cleaning things up.